### PR TITLE
fix(javascript): dependency imports and low-level init

### PR DESCRIPTION
**Pull request by shortcuts**
## 🧭 What and Why

🎟 JIRA Ticket: https://algolia.atlassian.net/browse/DI-2828

### Changes included:

closes https://github.com/algolia/algoliasearch-client-javascript/issues/1538

we import from the src of the dependency which is fine in the monorepo context, but the path isn't correct when importing from the built package

we actually don't need the low level init method, we can just use the regular instance and spread it, it should behave the same

### CODE REVIEW

File: clients/algoliasearch-client-javascript/base.rollup.config.js [hunk 1]
```diff
@@ -276,6 +276,7 @@ export function buildConfigs(pkg) {
           }),
           nodeResolve({
             preferBuiltins: true,
+            browser: isUmdBuild || isEsmBrowserBuild,
```
Comment (shortcuts) [inline clients/algoliasearch-client-javascript/base.rollup.config.js:279]: see https://www.npmjs.com/package/@rollup/plugin-node-resolve#browser, I should've set that way before D:

File: templates/javascript/clients/algoliasearch/builds/initClients.mustache [hunk 2]
```diff
@@ -1,61 +1,15 @@
 function initRecommend(initOptions: InitClientOptions = {}): RecommendClient {
-  return createRecommendClient({
-    ...commonOptions,
-    ...initOptions.options,
-    ...initOptions,
-  });
+  return recommendClient(initOptions.appId || appId, initOptions.apiKey ||apiKey, initOptions.options);
```
Comment (shortcuts) [inline templates/javascript/clients/algoliasearch/builds/initClients.mustache:2]: this is redundant and I think was legacy, now that standalone clients checks it, we don't need it here

File: clients/algoliasearch-client-javascript/base.rollup.config.js (hunk 1 above)
Comment (shortcuts) [reply clients/algoliasearch-client-javascript/base.rollup.config.js:279]: ok it's not fully working

**Review by millotp — APPROVED** [review]
looks good

**Pull request merged**